src/oid.rs

```diff
@@ -101,15 +101,12 @@ impl ObjectId {
         let counter = ObjectId::gen_count()?;
 
         let mut buf: [u8; 12] = [0; 12];
-        for i in 0..TIMESTAMP_SIZE {
-            buf[TIMESTAMP_OFFSET + i] = timestamp[i];
-        }
-        for i in 0..PROCESS_ID_SIZE {
-            buf[PROCESS_ID_OFFSET + i] = process_id[i];
-        }
-        for i in 0..COUNTER_SIZE {
-            buf[COUNTER_OFFSET + i] = counter[i];
-        }
+        buf[TIMESTAMP_OFFSET..(TIMESTAMP_SIZE + TIMESTAMP_OFFSET)]
```

Comment:
Apparently clippy is able to detect [manual copies of bytes](https://rust-lang.github.io/rust-clippy/master/index.html#manual_memcpy) and suggests to replace them with a method call, which is awesome